lighthouse-core/gather/gather-runner.js

```diff
@@ -301,6 +319,9 @@ class GatherRunner {
   static collectArtifacts(gathererResults) {
     const artifacts = {};
 
+    // Nest LighthouseRunWarnings, if any, so they will be collected into artifact.
+    gathererResults.LighthouseRunWarnings = [gathererResults.LighthouseRunWarnings || []];
```

Comment:
this ends up a bit weird. `gatherResults` expects an array of results for `[beforePass, pass, afterPass]` , the winner of which is selected just below. For adding warnings, however, that means everywhere a warning is added would have to do `gathererResults.LighthouseRunWarnings[0].push('...')`. This cheats it by keeping `gathererResults.LighthouseRunWarnings` a top level array everywhere (so it's just `gathererResults.LighthouseRunWarnings.push('...')`) and then nesting it into an array so it can be processed here